torch/csrc/autograd/autograd_meta.cpp

```diff
@@ -97,9 +97,23 @@ namespace {
     if (!at::_has_same_storage_numel(base, other)) {
       return false;
     }
+    if (base.is_conj() != other.is_conj()) {
+      return false;
+    }
```

Comment:
We'll need to do the same thing with `is_neg`, otherwise if you do `at::imag(a.conj())` we're going to run into similar errors I reckon.